pyro/infer/mcmc/mcmc.py

```diff
@@ -162,10 +167,23 @@ def _traces(self, *args, **kwargs):
                 if isinstance(val, Exception):
                     # Exception trace is already logged by worker.
                     raise val
-                elif val is not None:
-                    yield val
-                else:
+                results_buffer[chain_id - 1].append(val)
```

Comment:
It seems a little weird to `.append(None)` and then not yield it. Would it be simpler to rearrange so that you
```py
if val is None:
    active_worker -= 1
else:
    results_buffer[chain_id - 1].append(val)
    ...
```

Reply:
Thanks for the suggestion. The while loop can be outside, and this should work.